docs/control_tuple_support.md

```diff
@@ -0,0 +1,207 @@
+
+### DeliveryType IMMEDIATE
+As the name implies, the control tuple is immediately delivered to the next  downstream operator (if the operator is control aware), else it is forwarded to the next downstream operator.
+
+* **Case: Downstream is partitioned**  
+When the downstream is partitioned, the control tuple with *IMMEDIATE* delivery type would go to all the downstream partitions. This holds, irrespective of whether or not the control tuple was generated by the immediately upstream operator or even further upstream.
+
+* **Case: Upstream is partitioned**  
+When the upstream is partitioned and the control tuple is generated in any subset of the partitions the downstream operator would receive the control tuple immediately and would not wait till the end of the current window. In case the source for the control tuple was a single source further upstream and multiple copies were generated by the intermediate partitions, the duplicate copies of the control tuple would be filtered out at the downstream operator. Thus only unique control tuples are delivered to the downstream operator. Further, in case of *IMMEDIATE* delivery, the first instance of the control tuple is delivered to the operator and the duplicates filtered out.
+
+### DeliveryType END_WINDOW
+This delivery type only delivers the control tuple to the operator after all data tuples have been delivered to the operator. In the operator lifecycle, this would mean that the control tuples would be delivered just before the `endWindow` call.
+
+* **Case: Downstream is partitioned**  
+  When the downstream is partitioned, the control tuple emitted by the upstream would be broadcast to downstream operators and buffered in the downstream partitions until the end of the window and is delivered to the operator just before the `endWindow` call.
+
+* **Case: Upstream is partitioned**  
+  If the control tuples are generated in any subset of the partitions, then each control tuple is unique and are delivered to the downstream operator before the `endWindow` call. However, if the source for the control tuple is a source further upstream, then the downstream operator would filter out duplicates as and when each control tuple arrive at the operator, and finally all unique control tuples are delivered to the operator just before the `endWindow` call.
+## Assumptions
+All the user defined control tuples used in the application are cached in the memory of the operator for the duration of a window. For this reason, it is imperative that the size as well as the number of control tuples emitted within a window is small as compared to the number of data tuples.
+
+## Discussion Notes
+
+* [Requirements Discussion](https://lists.apache.org/thread.html/18225dd0142c42193cf2791d631092d5329c75a23b63246ee8232f0a@%3Cdev.apex.apache.org%3E) points to the requirements discussion for the feature.
```

Comment:
I would suggest to move these links into the JIRA and just provide the JIRA reference here.